numpy/_core/src/multiarray/number.c

```diff
@@ -363,7 +366,12 @@ fast_scalar_power(PyObject *o1, PyObject *o2, int inplace, PyObject **result)
     }
 
     PyArrayObject *a1 = (PyArrayObject *)o1;
-    if (!(PyArray_ISFLOAT(a1) || PyArray_ISCOMPLEX(a1))) {
+    if (PyArray_ISOBJECT(a1)) {
+        return 1;
+    }
+    if (!is_square && !PyArray_ISFLOAT(a1) && !PyArray_ISCOMPLEX(a1)) {
```

Comment:
Actually, sorry should have looked closer.  This is either square or sqrt.  Maybe we can just remove this branch also for `sqrt`?
Or would that change behavior drastically from the original version?  (I think I would lean towards that.

(If not, it may be nice to just use `fastop != nops.square` rather than introducing a new variable, but nitpicking.)

Reply:
Thanks for reviewing! It seems that in the old behavior only square was special-cased, as far as I can tell:

https://github.com/numpy/numpy/pull/27901/files#diff-1e638da588a4b86ad9e4c672ae59fa63c86a5232711d0a9f1a829ae89010d54aL464

It's a bit confusing, so would have been nice to have gotten rid of it, but well :)

> (If not, it may be nice to just use fastop != nops.square rather than introducing a new variable, but nitpicking.)
Actually yes, let me remove the variable, thanks!